src/mender-update/daemon/state_machine/state_machine.cpp

```diff
@@ -210,6 +213,7 @@ StateMachine::StateMachine(Context &ctx, events::EventLoop &event_loop) :
 
 	main_states_.AddTransition(send_commit_status_state_,               se::Success,                     ss.commit_enter_,                        tf::Immediate);
 	main_states_.AddTransition(send_commit_status_state_,               se::Failure,                     update_check_rollback_state_,            tf::Immediate);
+	main_states_.AddTransition(send_commit_status_state_,               se::DeploymentAborted,           update_check_rollback_state_,            tf::Immediate);
```

Comment:
And from here on, I would add a comment with the opposite meaning. Once we are committing and after we have commited the generic failure and the abort are treated the same way.